ext/afform/core/ang/af/afField.component.js

```diff
@@ -279,6 +279,10 @@
       // Getter/Setter function for most fields (except select & entityRef)
       $scope.getSetValue = function(val) {
         var currentVal = $scope.dataProvider.getFieldData()[ctrl.fieldName];
+        // boolean custom fields uses 0/1 as radio button values
+        if (ctrl.defn.data_type === 'Boolean' && currentVal !== undefined && ctrl.fieldName.includes('.')) {
```

Comment:
What is the rationale for `ctrl.fieldName.includes('.')`? To target custom fields specifically?
APIv4 is supposed to treat all fields the same whether they are custom or not (and if it doesn't, that's a bug).

Reply:
This is clearly a bug... but I don't know how to better fix it :)

Html input for bool custom field (i.e. `value="1"`):
```
<input class="crm-form-radio ng-pristine ng-untouched ng-valid ng-valid-required ng-not-empty" name="constituent-information-testbool-0" type="radio" ng-model="getSetValue" ng-required="$ctrl.defn.required" ng-model-options="{getterSetter: true}" ng-value="opt.id" value="1">
```

Html input for bool core field (i.e. `value="true"`):
```
<input class="crm-form-radio ng-pristine ng-untouched ng-valid ng-valid-required ng-not-empty" name="is-opt-out-1" type="radio" ng-model="getSetValue" ng-required="$ctrl.defn.required" ng-model-options="{getterSetter: true}" ng-value="opt.id" value="true">
```